client/clientservice/src/request_service.cpp

```diff
@@ -119,12 +119,16 @@ void RequestServiceCallData::sendToConcordClient() {
           break;
         case (static_cast<uint32_t>(bftEngine::OperationResult::CONFLICT_DETECTED)):
           LOG_INFO(logger, "Request failed with CONFLICT_DETECTED error for cid=" << req_config.correlation_id);
-          status = grpc::Status(grpc::StatusCode::ABORTED, "Aborted");
+          status = grpc::Status(grpc::StatusCode::ABORTED, "Aborted due to conflict detected");
           break;
         case (static_cast<uint32_t>(bftEngine::OperationResult::OVERLOADED)):
           LOG_INFO(logger, "Request failed with OVERLOADED error for cid=" << req_config.correlation_id);
           status = grpc::Status(grpc::StatusCode::RESOURCE_EXHAUSTED, "All clients occupied");
           break;
+        case (static_cast<uint32_t>(bftEngine::OperationResult::OUT_OF_TIME_BOUND_ERROR)):
+          LOG_INFO(logger, "Request failed with OUT_OF_TIME_BOUND_ERROR error for cid=" << req_config.correlation_id);
+          status = grpc::Status(grpc::StatusCode::ABORTED, "Aborted due to out of time bound error");
```

Comment:
I've asked Celine this question offline and proposed to sync with you regarding it.

Reply:
Timeout error is different and it is was request time out.
OUT_OF_TIME_BOUND_ERROR is an error from commandhandler like CONFLICT_DETECTED. Previously explicit write sets were provided by the execution engine if the recording time is not within time bounds. Now this is a new change in behaviour where errors have to be send back in such cases which will propagate as ABORT to the application so that they can choose to act based on this error.

Reply:
To my knowledge, Concord does not check or enforce a time span in which the result of a request can be persisted. This is execution engine specific. Therefore, I don't see a reason to introduce a new error type which doesn't relate to Concord's behavior. If the execution engine doesn't want to persist a result then it can signal this via an application specific response.

Reply:
Post discussion, there is requirement to distinguish the ABORTED error due to conflict detected and rejection during processing. 
As discussed, I have updated changes to make it generic rejection error instead of OUT_OF_TIME_BOUND_ERROR